src/transformStateWithClones.js

```diff
@@ -6,8 +6,45 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const arrayWithNewStates = [];
+
+  for (const action of actions) {
+    transform(newState, action);
+
+    const savedNewState = { ...newState };
+
+    arrayWithNewStates.push(savedNewState);
```

Comment:
let's try to make this code shorter and without redundant variables 
```suggestion
    transform(newState, action);
    arrayWithNewStates.push({ ...newState });
```